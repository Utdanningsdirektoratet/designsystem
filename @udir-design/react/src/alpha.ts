/*
 * This is the entry point for the /alpha sub-module.
 * It exports the alpha-stage components, plus everything in beta (& stable).
 */
export * from './beta';
export * from './components/alpha';
export * from './utilities/hooks/alpha';
