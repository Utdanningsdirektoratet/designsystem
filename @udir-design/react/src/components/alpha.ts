/*
 * This file exports the components that are in the alpha stage of development
 */

export * from './button/Button';
export * from './alert/Alert';
export * from './avatar/Avatar';
export * from './badge/Badge';
export * from './breadcrumbs/Breadcrumbs';
export * from './card/Card';
export * from './checkbox/Checkbox';
export * from './chip/Chip';
export * from './details/Details';
export * from './dialog/Dialog';
export * from './divider/Divider';
export * from './dropdown/Dropdown';
export * from './errorSummary/ErrorSummary';
export * from './field/Field';
export * from './fieldset/Fieldset';
export * from './input/Input';
export * from './link/Link';
export * from './list/List';
export * from './loaders';
export * from './pagination/Pagination';
export * from './popover/Popover';
export * from './radio/Radio';
export * from './search/Search';
export * from './select/Select';
export * from './skipLink/SkipLink';
export * from './switch/Switch';
export * from './table/Table';
export * from './tabs/Tabs';
export * from './tag/Tag';
export * from './textarea/Textarea';
export * from './textfield/Textfield';
export * from './toggleGroup/ToggleGroup';
export * from './tooltip/Tooltip';
export * from './typography';
