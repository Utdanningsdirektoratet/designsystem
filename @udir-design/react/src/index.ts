export * from './button/Button';
export * from './accordion/Accordion';
export * from './alert/Alert';
export * from './avatar/Avatar';
export * from './badge/Badge';
export * from './breadcrumbs/Breadcrumbs';
export * from './card/Card';
export * from './checkbox/Checkbox';
export * from './chip/Chip';
export * from './divider/Divider';
export * from './dropdown/Dropdown';
export * from './error-summary/ErrorSummary';
export * from './fieldset/Fieldset';
export * from './help-text/HelpText';
export * from './link/Link';
export * from './list/List';
export * from './loaders/index';
export * from './modal/Modal';
export * from './pagination/Pagination';
export * from './popover/Popover';
export * from './radio/Radio';
export * from './search/Search';
export * from './select/Select';
export * from './skip-link/SkipLink';
export * from './switch/Switch';
export * from './table/Table';
export * from './tabs/Tabs';
export * from './tag/Tag';
