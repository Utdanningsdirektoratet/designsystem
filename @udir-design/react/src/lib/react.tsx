import styles from './react.module.scss';

export const React: React.FC = () => (
  <div className={styles['container']}>
    <h1>Welcome to React!</h1>
  </div>
);

export default React;
