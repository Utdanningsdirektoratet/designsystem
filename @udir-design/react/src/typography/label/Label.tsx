export { Label, type LabelProps } from '@digdir/designsystemet-react';
