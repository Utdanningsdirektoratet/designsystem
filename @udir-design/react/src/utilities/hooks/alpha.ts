/*
 * This file exports the hooks that are in the alpha stage of development
 */

export {
  type UsePaginationProps,
  usePagination,
} from './usePagination/usePagination';
export {
  useTableOfContents,
  type useTableOfContentsProps,
} from './useTableOfContents/useTableOfContents';
