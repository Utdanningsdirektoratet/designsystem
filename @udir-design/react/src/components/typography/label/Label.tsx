import {
  Label as DigdirLabel,
  type LabelProps as DigdirLabelProps,
} from '@digdir/designsystemet-react';
import { ComponentRef, ForwardRefExoticComponent, RefAttributes } from 'react';

type LabelProps = Omit<DigdirLabelProps, 'data-color'>;

const Label = DigdirLabel as ForwardRefExoticComponent<
  LabelProps & RefAttributes<ComponentRef<typeof DigdirLabel>>
>;

export { Label, LabelProps };
