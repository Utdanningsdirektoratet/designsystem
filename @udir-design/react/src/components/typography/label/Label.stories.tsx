import type { Meta, StoryObj } from '@storybook/react-vite';
import { Label } from './Label';

const meta: Meta<typeof Label> = {
  component: Label,
  tags: ['beta', '!autodocs', 'digdir'],
  parameters: {
    componentOrigin: {
      originator: 'digdir',
    },
  },
  title: 'Components/Typography/Label',
};

export default meta;
type Story = StoryObj<typeof Label>;

export const Preview: Story = {
  args: {
    children: 'Vennligst skriv inn f√∏dselsnummer. 11 tegn',
    weight: 'medium',
  },
};
