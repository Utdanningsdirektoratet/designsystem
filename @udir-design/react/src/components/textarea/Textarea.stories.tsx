import type { Meta, StoryObj } from '@storybook/react';
import { Label } from '../typography/label/Label';
import { useState } from 'react';
import { Button, Divider, Paragraph, Textarea } from '../alpha';
import { expect, userEvent, within } from '@storybook/test';

const meta: Meta<typeof Textarea> = {
  component: Textarea,
  tags: ['alpha'],
  parameters: {
    customStyles: {
      display: 'flex',
      flexDirection: 'column',
      gap: 'var(--ds-size-2)',
      maxWidth: '20rem',
    },
  },
};

export default meta;
type Story = StoryObj<typeof Textarea>;

export const Preview: Story = {
  args: {
    disabled: false,
    readOnly: false,
    id: 'my-textarea',
  },
  render: (args) => (
    <>
      <Label htmlFor={args.id}>Label</Label>
      <Textarea {...args} />
    </>
  ),
  play: async ({ canvasElement, step, args }) => {
    const canvas = within(canvasElement);
    const textarea = canvas.getByRole('textbox');

    await step('Textarea is rendered', async () => {
      expect(textarea).toBeInTheDocument();
    });

    await step('User can focus the textarea', async () => {
      await userEvent.click(textarea);
      expect(textarea).toHaveFocus();
    });

    await step('User can blur the textarea', async () => {
      await userEvent.tab();
      expect(textarea).not.toHaveFocus();
    });

    if (!args.disabled && !args.readOnly) {
      await step('User can type in the textarea', async () => {
        await userEvent.clear(textarea);
        await userEvent.type(textarea, 'Hello World');
        expect(textarea).toHaveValue('Hello World');
      });
    } else {
      await step(
        'Textarea is disabled or read-only so it should not be editable',
        async () => {
          expect(textarea).toBeDisabled();
        },
      );
    }
  },
};

export const FullWidth: Story = {
  args: {
    id: 'my-textarea',
  },
  parameters: {
    customStyles: {
      maxWidth: '100%',
    },
  },
  render: (args) => (
    <>
      <Label htmlFor={args.id}>Label</Label>
      <Textarea {...args} />
    </>
  ),
};

export const Controlled: Story = {
  args: {
    id: 'my-textarea',
  },
  render: function Render(args) {
    const [value, setValue] = useState(`${args.value || ''}`);

    return (
      <>
        <Label htmlFor={args.id}>Kontroller meg!</Label>
        <Textarea
          onChange={(e) => setValue(e.target.value)}
          value={value}
          {...args}
        />

        <Divider style={{ marginTop: 'var(--ds-size-4)' }} />

        <Paragraph style={{ margin: 'var(--ds-size-2) 0' }}>
          Du har skrevet inn: {value}
        </Paragraph>
        <Button onClick={() => setValue('Pizza')}>Jeg vil ha Pizza</Button>
      </>
    );
  },
};
