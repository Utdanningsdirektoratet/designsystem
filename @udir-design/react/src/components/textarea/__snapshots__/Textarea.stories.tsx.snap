// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Controlled 1`] = `
<label
  data-weight="medium"
  for="textfield-controlled"
>
  Beskriv bekymringen din
</label>
<textarea
  id="textfield-controlled"
  rows="4"
>
</textarea>
<button
  data-variant="secondary"
  type="button"
  style="margin-top: var(--ds-size-2);"
>
  TÃ¸m feltet
</button>
`;

exports[`Full Width 1`] = `
<label
  data-weight="medium"
  for="my-textarea"
>
  Beskriv bekymringen din
</label>
<textarea id="my-textarea">
</textarea>
`;

exports[`Preview 1`] = `
<label
  data-weight="medium"
  for="my-textarea"
>
  Label
</label>
<textarea
  rows="3"
  cols="20"
  id="my-textarea"
>
</textarea>
`;
